pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ConcurrentMapPartitionUpsertMetadataManager.java

```diff
@@ -307,7 +309,7 @@ protected GenericRow doUpdateRecord(GenericRow record, RecordInfo recordInfo) {
             int currentDocId = recordLocation.getDocId();
             if (currentQueryableDocIds == null || currentQueryableDocIds.contains(currentDocId)) {
               _reusePreviousRow.init(currentSegment, currentDocId);
-              _partialUpsertHandler.merge(_reusePreviousRow, record);
+              _partialUpsertHandler.merge(_reusePreviousRow, record, _reuseMergerResult);
```

Comment:
I feel this reusable map should be maintained within the `PartialUpsertHandler`. I don't see why it should be part of the interface

Reply:
The`_reuseMergerResult` map is defined in the context of a partition for a consuming segment. If we move it to `PartialUpsertHandler` then we need to modify `PartialUpsertHandler` to be initialised in `BasePartitionUpsertMetadataManager` instead of `BaseTableUpsertMetadataManager`. Moreover the `_reuseMergerResult` would have to be made threadsafe across consuming segments.